client/src/app/api/post-admin-login/route.ts

```diff
@@ -0,0 +1,47 @@
+import {NextResponse} from 'next/server';
+import {postAdminLogin} from '@apis/admin';
+import {cookies} from 'next/headers';
+
+export const POST = async (request: Request) => {
+  try {
+    const {loginId, password} = await request.json();
+
+    if (!loginId || !password) {
+      return NextResponse.json({error: 'Login ID 또는 password가 존재하지 않습니다.'}, {status: 400});
+    }
+
+    const response = await postAdminLogin({loginId, password});
+
+    if (!response.ok) {
+      return NextResponse.json({error: '로그인 실패'}, {status: 401});
+    }
```

Comment:
`response.ok` 가 false인 경우는 어떤 것들이 있을까요?
401 코드는 보통 client의 인증 정보가 잘못 된 경우에 반환하는 것 같아서, 500코드가 어울리지 않을까 싶었어요
하지만, 여기서 호출하는 API가 우리 서버의 api다보니....  이 안에서도 다양한 오류 케이스가 있겠다 싶었어요
리바이가 에러 케이스를 세분화 해서 잘 나눠준 걸 보니... 저희가 에러 관리를 너무 안해놓았다 싶긴 하네요... 하하...

```suggestion
    if (!response.ok) {
      const errorStatus = response.status >= 500 ? 500 : response.status;
      const errorMessage = errorStatus >= 500 ? '서버 오류가 발생했습니다.' : '로그인에 실패했습니다.';
      return NextResponse.json({ error: errorMessage }, { status: errorStatus });
    }
```

Reply:
피드백 감사합니다! 저도 코드짜면서 id와 password가 문제가 있으면 사실 `client`측 문제이지만 post하는 곳은 next.js 안에서 `route.ts`이기 때문에 `server`이기도 해서..  이 에러로직에 대해서 어떻게 해야할까 많이 고민했었네요.
진짜 상세히 나누자면 아래와 같이 더 나눌 수 있는데 어느 정도가 나은지 모르겠네요..!

```tsx
export const POST = async (request: Request) => {
  try {
    const { loginId, password } = await request.json();

    if (!loginId || !password) {
      return NextResponse.json(
        { error: 'Login ID 또는 password가 존재하지 않습니다.' },
        { status: 400 }
      );
    }

    const response = await postAdminLogin({ loginId, password });

    if (!response.ok) {
      // 상태 코드별로 구분 처리
      switch (response.status) {
        case 400:
          return NextResponse.json(
            { error: '잘못된 요청입니다. 입력 값을 확인해주세요.' },
            { status: 400 }
          );
        case 401:
          return NextResponse.json(
            { error: '아이디 또는 비밀번호가 올바르지 않습니다.' },
            { status: 401 }
          );
        case 403:
          return NextResponse.json(
            { error: '접근 권한이 없습니다.' },
            { status: 403 }
          );
        case 500:
          return NextResponse.json(
            { error: '서버 오류가 발생했습니다. 잠시 후 다시 시도해주세요.' },
            { status: 500 }
          );
        default:
          return NextResponse.json(
            { error: '알 수 없는 오류가 발생했습니다.' },
            { status: response.status }
          );
      }
    }

    const setCookieHeader = response.headers.get('set-cookie');

    if (!setCookieHeader) {
      return NextResponse.json(
        { error: '로그인 토큰이 없습니다.' },
        { status: 401 }
      );
    }

    const [tokenKey, rest] = setCookieHeader.split('=');
    const tokenValue = rest?.split(';')[0];

    if (!tokenKey || !tokenValue) {
      return NextResponse.json(
        { error: '로그인 토큰이 유효하지 않습니다.' },
        { status: 401 }
      );
    }

    const cookieStore = await cookies();
    cookieStore.set({
      name: tokenKey.trim(),
      value: tokenValue.split(';')[0].trim(),
      httpOnly: true,
      secure: process.env.NODE_ENV === 'production',
      sameSite: 'strict',
      maxAge: 60 * 60 * 24,
      path: '/',
    });

    return NextResponse.json({ message: '로그인 성공' });
  } catch (error) {
    console.error('로그인 처리 중 서버 오류:', error);
    return NextResponse.json(
      { error: '로그인 처리 중 오류' },
      { status: 500 }
    );
  }
};

```